notes/trees.md

```diff
@@ -0,0 +1,234 @@
+# Trees.scala
+
+## Top level (`meta.Source`, `meta.Pkg`, `meta.Pkg`)
+
+```scala
+"package a".parse[Source].get.structure
+```
+
+## Declarations (`meta.Decl`)
+
+```scala
+"val a: Int".parse[Stat].get.structure
+```
+
+Decl            | Examples
+----------------|-----------
+Decl.Val        | val a
```

Comment:
Can we include a `: Int` type annotations here? The ability to leave out `: Unit` here is a deprecated feature